docs/modules/library-development/pages/classes.adoc

```diff
@@ -0,0 +1,63 @@
+# Create a Class
+
+## Library Structure
+
+Classes should be placed in the `src` directory. It is recommended that a package be used and named after the library contributing the class.
+
+[TIP]
+====
+Each loaded library's `src` directory contents are synced to a common directory.  Colliding file paths will result in an exception being thrown.
+====
+
+## Serializability
+
+In order to Jenkins pipelines to resume, a design pattern called Continuation Passing Style (CPS) has been implemented in the Pipeline Plugin. CPS works by continuously serializing the current state of the pipeline to disk such that if the pipeline were to be paused or Jenkins restarted, the pipeline could resume where it left off.
+
+Therefore, every variable that's persisted in the pipeline's context must be serializable.
+
+To help ensure that the classes that are contributed by a library are serializable, they should implement the `Serializable` interface.
+
+For example, a `Utility` class coming from an `example` library:
+
+[source, groovy]
+----
+package example
+
+class Utility implements Serializable {}
+----
+
+[NOTE]
+====
+To learn more, check out https://www.jenkins.io/doc/book/pipeline/pipeline-best-practices/#avoiding-notserializableexception[Best Pracies for Avoiding Serializability Exceptions]
+====
+
+## Accessibility
+
+Classes contributed by loaded libraries can be import from the pipeline template, steps of the same library, and steps from other libraries.
```

Comment:
Classes contributed by loaded libraries can be import from -> 'Classes contributed by loaded libraries can be *imported into*'